support/azure/azure-kubernetes/availability-performance/node-auto-repair-errors.md

```diff
@@ -0,0 +1,42 @@
+
+During the node auto-repair process, AKS will initiate reboot, reimage, and redeploy actions on your unhealthy node. These repair actions may encounter errors due to various underlying causes. The resulting error code is surfaced through a [Kubernetes event](/azure/aks/events), which you can use to monitor the status of your node and auto-repair actions. This article discusses common errors, together with their potential causes and next steps, as well as best practices for monitoring node auto-repair.
+
+## Prerequisites
+To determine what type of node auto-repair error has occured, look for one of the following Kubernetes events: 
+| Reason | Event Message | Description |
+| --- | --- | --- |
+| NodeRebootError | Node auto-repair reboot action failed due to an operation failure: [error code here] | Emitted when there is an error with the reboot action. |
+| NodeReimageError | Node auto-repair reimage action failed due to an operation failure: [error code here] | Emitted when there is an error with the reimage action. |
+| NodeRedeployError | Node auto-repair redeploy action failed due to an operation failure: [error code here] | Emitted when there is an error with the redeploy action. |
+
+> [!NOTE]
+> In most cases, node auto-repair errors will not cause any impact to your cluster or applications since your node was already in an unhealthy state prior to the auto-repair process. When encountering node auto-repair errors, we recommend that you first attempt to repair the node yourself by following the instructions here: [Basic troubleshooting of Node Not Ready failures](./node-not-ready-basic-troubleshooting.md). If you are unable to restore your node back to a Succeeded status and are noticing persistent errors reported by node auto-repair, you may be encountering a rare issue and should reach out to Azure support.
+
+## Common error codes
+The table below contains the most common node auto-repair errors.
+
+| Error code | Causes & Solution |
+|---|---|
+| VMExtensionProvisioningError | One or more VM extensions failed to be provisioned on the VM. Read more on possible error types and troubleshooting steps at [Troubleshoot the ERR_VHD_FILE_NOT_FOUND error code (124)](../create-upgrade-delete/error-code-vhdfilenotfound.md). To determine the exact VM extension provisioning error on your node, read more at [View error details in the Azure portal](../create-upgrade-delete/troubleshoot-aks-cluster-creation-issues.md#view-resources-in-the-azure-portal). |
+| InvalidParameter | This is an internal error which occurs when the node auto-repair process requests a VM that no longer exists. If you notice any new symptoms or interruptions on your node, please reach out to Azure support. |
+| scaleSetNameAndInstanceIDFromProviderID failed | This is an internal issue where the node may not have been provisioned correctly. If you notice any new symptoms or interruptions on your node, please reach out to Azure support. |
```

Comment:
Just remembered this and "ManagedIdentityCredential authentication failed" are not going to show up unless the customer is seeing a Diagnose and Solve detector with remediator service logs, since these are not ARM error codes. We don't have that right now.